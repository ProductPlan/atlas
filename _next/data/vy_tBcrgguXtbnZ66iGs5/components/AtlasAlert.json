{"pageProps":{"component":"AtlasAlert"},"__N_SSG":true}