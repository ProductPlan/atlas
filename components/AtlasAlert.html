<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><title>ProductPlan Atlas</title><meta name="next-head-count" content="3"/><link rel="preconnect" href="https://fonts.googleapis.com"/><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin="true"/><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.1.2/css/all.min.css"/><link rel="shortcut icon" href="/atlas/favicon.ico"/><link rel="preconnect" href="https://fonts.gstatic.com" crossorigin /><link rel="preload" href="/atlas/_next/static/css/0ed7519f8f6e2b12.css" as="style"/><link rel="stylesheet" href="/atlas/_next/static/css/0ed7519f8f6e2b12.css" data-n-g=""/><link rel="preload" href="/atlas/_next/static/css/48309c2d926084e7.css" as="style"/><link rel="stylesheet" href="/atlas/_next/static/css/48309c2d926084e7.css" data-n-p=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/atlas/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js"></script><script src="/atlas/_next/static/chunks/webpack-e0e074d834bfaea7.js" defer=""></script><script src="/atlas/_next/static/chunks/framework-c02f198d58d34563.js" defer=""></script><script src="/atlas/_next/static/chunks/main-88e61d9f66dc3fee.js" defer=""></script><script src="/atlas/_next/static/chunks/pages/_app-b2c4922735fa1d06.js" defer=""></script><script src="/atlas/_next/static/chunks/874-c64678d9ac232da4.js" defer=""></script><script src="/atlas/_next/static/chunks/917-19d7d6836e5ffd88.js" defer=""></script><script src="/atlas/_next/static/chunks/95-8c388448e411c861.js" defer=""></script><script src="/atlas/_next/static/chunks/pages/components/%5Bcomponent%5D-7680a099196e1d13.js" defer=""></script><script src="/atlas/_next/static/oSPjchGiWCh82QQls6uKg/_buildManifest.js" defer=""></script><script src="/atlas/_next/static/oSPjchGiWCh82QQls6uKg/_ssgManifest.js" defer=""></script><style data-href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;700&display=swap">@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuLyfMZs.woff) format('woff')}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuI6fMZs.woff) format('woff')}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcCO3FwrK3iLTeHuS_fvQtMwCp50KnMw2boKoduKmMEVuFuYMZs.woff) format('woff')}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0460-052F,U+1C80-1C88,U+20B4,U+2DE0-2DFF,U+A640-A69F,U+FE2E-FE2F}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0301,U+0400-045F,U+0490-0491,U+04B0-04B1,U+2116}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+1F00-1FFF}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0370-03FF}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0102-0103,U+0110-0111,U+0128-0129,U+0168-0169,U+01A0-01A1,U+01AF-01B0,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1EA0-1EF9,U+20AB}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0100-02AF,U+0304,U+0308,U+0329,U+1E00-1E9F,U+1EF2-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF}@font-face{font-family:'Inter';font-style:normal;font-weight:400;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0304,U+0308,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0460-052F,U+1C80-1C88,U+20B4,U+2DE0-2DFF,U+A640-A69F,U+FE2E-FE2F}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0301,U+0400-045F,U+0490-0491,U+04B0-04B1,U+2116}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+1F00-1FFF}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0370-03FF}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0102-0103,U+0110-0111,U+0128-0129,U+0168-0169,U+01A0-01A1,U+01AF-01B0,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1EA0-1EF9,U+20AB}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0100-02AF,U+0304,U+0308,U+0329,U+1E00-1E9F,U+1EF2-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF}@font-face{font-family:'Inter';font-style:normal;font-weight:500;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0304,U+0308,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0460-052F,U+1C80-1C88,U+20B4,U+2DE0-2DFF,U+A640-A69F,U+FE2E-FE2F}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0301,U+0400-045F,U+0490-0491,U+04B0-04B1,U+2116}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+1F00-1FFF}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0370-03FF}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0102-0103,U+0110-0111,U+0128-0129,U+0168-0169,U+01A0-01A1,U+01AF-01B0,U+0300-0301,U+0303-0304,U+0308-0309,U+0323,U+0329,U+1EA0-1EF9,U+20AB}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');unicode-range:U+0100-02AF,U+0304,U+0308,U+0329,U+1E00-1E9F,U+1EF2-1EFF,U+2020,U+20A0-20AB,U+20AD-20CF,U+2113,U+2C60-2C7F,U+A720-A7FF}@font-face{font-family:'Inter';font-style:normal;font-weight:700;font-display:swap;src:url(https://fonts.gstatic.com/s/inter/v13/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');unicode-range:U+0000-00FF,U+0131,U+0152-0153,U+02BB-02BC,U+02C6,U+02DA,U+02DC,U+0304,U+0308,U+0329,U+2000-206F,U+2074,U+20AC,U+2122,U+2191,U+2193,U+2212,U+2215,U+FEFF,U+FFFD}</style></head><body><div id="__next"><div><div class="Heading_Heading__EQADp"><a href="/atlas">üåê ProductPlan Atlas</a></div><div class="Documentation"><nav class="Navigation_Sidebar__ycYDm"><div class="Navigation_Sidebar__title__LoJn8">Atlas</div><ul class="Navigation_Sidebar__list__Nlfdq"><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas">Welcome</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="https://github.com/ProductPlan/atlas-mono" target="_blank" rel="noreferrer">GitHub <i class="fa-brands fa-github"></i></a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/tokens">Tokens</a></li></ul><div class="Navigation_Sidebar__title__LoJn8">Components</div><ul class="Navigation_Sidebar__list__Nlfdq"><li class="Navigation_Sidebar__listItem__5Wet1 Navigation_Sidebar__listItemActive__3t7we"><a href="/atlas/components/AtlasAlert">AtlasAlert</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/components/AtlasBanner">AtlasBanner</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/components/AtlasPopover">AtlasPopover</a></li><li class="Navigation_Sidebar__listItem__5Wet1"><a href="/atlas/components/BasicButton">BasicButton</a></li></ul></nav><div class="PageContent_Page__KQX_Q"><div class="PageContent_PageContent__Hwrl_"><h1 class="PageContent_PageHeading__H2sy3">AtlasAlert</h1><p class="PageContent_PageSubcopy__Ux2Ic">An alert which contains important information about a page.</p><div><h3>Example<!-- -->s</h3><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Alert Types</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div style="border:2px solid var(--a-color-light-gray-100);width:556px;border-radius:4px"><div style="margin:4px"><div class="AtlasAlert AtlasAlert--default"><i class="fas fa-circle-info"></i><div class="AtlasAlert__body"><div class="AtlasAlert__content">This is a default alert.</div></div></div></div><div style="margin:0px 4px 4px 4px"><div class="AtlasAlert AtlasAlert--information"><i class="fas fa-circle-info"></i><div class="AtlasAlert__body"><div class="AtlasAlert__content">This is an information alert.</div></div></div></div><div style="margin:0px 4px 4px 4px"><div class="AtlasAlert AtlasAlert--success"><i class="fas fa-circle-check"></i><div class="AtlasAlert__body"><div class="AtlasAlert__content">This is a success alert.</div></div></div></div><div style="margin:0px 4px 4px 4px"><div class="AtlasAlert AtlasAlert--warning"><i class="fas fa-circle-exclamation"></i><div class="AtlasAlert__body"><div class="AtlasAlert__content">This is a warning alert.</div></div></div></div><div style="margin:0px 4px 4px 4px"><div class="AtlasAlert AtlasAlert--danger"><i class="fas fa-triangle-exclamation"></i><div class="AtlasAlert__body"><div class="AtlasAlert__content">This is a danger alert.</div></div></div></div></div></div><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Alert with Button</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div style="border:2px solid var(--a-color-light-gray-100);width:556px;border-radius:4px"><div style="margin:4px"><div class="AtlasAlert AtlasAlert--default"><i class="fas fa-circle-info"></i><div class="AtlasAlert__body"><div class="AtlasAlert__content">This is the content of a default alert.</div><atlas-basic-button align="center" label="Default" size="mini" theme="default" type="button"></atlas-basic-button></div></div></div></div></div><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Alert with Title</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div style="border:2px solid var(--a-color-light-gray-100);width:556px;border-radius:4px"><div style="margin:4px"><div class="AtlasAlert AtlasAlert--information"><i class="fas fa-circle-info"></i><div class="AtlasAlert__body"><div class="AtlasAlert__title">This information alert has a title!</div><div class="AtlasAlert__content">This is the content of an information alert with only a title.</div></div></div></div></div></div><div class="ComponentPage_Example__VsPS1"><header class="ComponentPage_ExampleHeader__jLmLE"><div><h4>Alert with Button and Title</h4></div><div><button class="IconButton"><i class="fa fa-code"></i></button></div></header><div style="border:2px solid var(--a-color-light-gray-100);width:556px;border-radius:4px"><div style="margin:4px"><div class="AtlasAlert AtlasAlert--success"><i class="fas fa-circle-check"></i><div class="AtlasAlert__body"><div class="AtlasAlert__title">This success alert has a title!</div><div class="AtlasAlert__content">This is the content of a success alert that has a title and button.</div><atlas-basic-button align="center" label="Success" size="mini" theme="default" type="button"></atlas-basic-button></div></div></div></div></div><h3>Props</h3><table class="Table"><thead><tr><th class="Props_TableHeading__Kvchn">Name</th><th class="Props_TableHeading__Kvchn">Description</th><th class="Props_TableHeading__Kvchn">Type</th><th class="Props_TableHeading__Kvchn">Default</th><th class="Props_TableHeading__Kvchn">Required</th></tr></thead><tbody><tr><td class="Props_TableCell__zHnLs"><strong>type</strong></td><td class="Props_TableCell__zHnLs">one of default, information, success, warning, danger</td><td class="Props_TableCell__zHnLs">enum</td><td class="Props_TableCell__zHnLs">&quot;default&quot;</td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>title</strong></td><td class="Props_TableCell__zHnLs">text to display for the title</td><td class="Props_TableCell__zHnLs">string</td><td class="Props_TableCell__zHnLs">null</td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>label</strong></td><td class="Props_TableCell__zHnLs">text to display in the button</td><td class="Props_TableCell__zHnLs">string</td><td class="Props_TableCell__zHnLs">null</td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>handleClick</strong></td><td class="Props_TableCell__zHnLs">click handler</td><td class="Props_TableCell__zHnLs">func</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>content</strong></td><td class="Props_TableCell__zHnLs">text to display for the content</td><td class="Props_TableCell__zHnLs">string</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"><i class="fa fa-check-circle"></i></td></tr><tr><td class="Props_TableCell__zHnLs"><strong>icon</strong></td><td class="Props_TableCell__zHnLs">icon to display to the left of alert content</td><td class="Props_TableCell__zHnLs">element</td><td class="Props_TableCell__zHnLs"></td><td class="Props_TableCell__zHnLs Props_TableCellCenter__zg1dd"><i class="fa fa-check-circle"></i></td></tr></tbody></table></div></div></div></div></div></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"component":"AtlasAlert"},"__N_SSG":true},"page":"/components/[component]","query":{"component":"AtlasAlert"},"buildId":"oSPjchGiWCh82QQls6uKg","assetPrefix":"/atlas","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>