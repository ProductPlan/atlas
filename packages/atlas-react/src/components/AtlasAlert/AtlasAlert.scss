@import "~@productplan/atlas-styles/atlas.scss";

.AtlasAlert{
  border-radius: 4px;
  display: flex;
  flex-direction: row;

  &__body {
    margin:  16px 16px 16px 8px;
    font-size: 14px;
    line-height: 20px;
    & > * {
      margin-bottom: 4px;
    }
    & > *:last-child {
      margin-bottom: 0px;
    }
  }
  
  &__title {
    font-weight: var(--a-font-weight-font-500);
  }
  
  &__content {
    font-weight:var(--a-font-weight-font-400);
    margin-bottom: 8px;
  }

  i {
    margin-top: 16px;
    margin-left: 16px;
    width: 20px;
    height: 20px;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  &--default {
    background: var(--a-color-light-gray-100);
    i {
      color: var(--a-color-gray-400)
    }

    .BasicButton {
      background-color: var(--a-color-gray-400)
    }
  }
  &--information{
    background: var(--a-color-blue-100);
    color: var(--a-color-blue-700);
    i {
      color: var(--a-color-blue-400)
    }
  }
  &--success {
    background: var(--a-color-green-100);
    color: var(--a-color-green-700);
    i {
      color: var(--a-color-green-400)
    }
  }
  &--warning {
    background: var(--a-color-yellow-100);
    color: var(--a-color-yellow-700);
    i {
      color: var(--a-color-yellow-400)
    }
  }
  &--danger {
    background: var(--a-color-red-100);
    color: var(--a-color-red-700);
    i {
      color: var(--a-color-red-400)
    }
  }
}



