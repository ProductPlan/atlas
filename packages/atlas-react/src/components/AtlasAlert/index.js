export { default, alertTypes } from "./AtlasAlert";
