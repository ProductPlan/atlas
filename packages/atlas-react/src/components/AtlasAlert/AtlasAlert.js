import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import BasicButton from "@components/BasicButton";
import "./AtlasAlert.scss";

export const alertTypes = ["default", "information", "success", "warning", "danger"];
/** An alert which contains important information about a page.  */
export default function AtlasAlert({
  type = "default",
  title=null,
  label=null,
  handleClick,
  content,
  icon,
}) {
  const classes = classNames("AtlasAlert", { [`AtlasAlert--${type}`]: true });

  return (
    <div className={classes}>
      {icon}
      <div className="AtlasAlert__body">
        {title && <div className="AtlasAlert__title">{title}</div>}
        <div className="AtlasAlert__content">{content}</div>
        {label && (
          <BasicButton
            className="AtlasAlert__button"
            label={label}
            size="mini"
            onClick={handleClick}
          />
        )}
      </div>
    </div>
  );
}

AtlasAlert.propTypes = {
  /** one of default, information, success, warning, danger */
  type: PropTypes.oneOf(alertTypes),
  /** text to display for the title */
  title: PropTypes.string,
  /** text to display in the button */
  label: PropTypes.string,
  /** click handler */
  handleClick: PropTypes.func,
  /** text to display for the content */
  content: PropTypes.string.isRequired,
  /** icon to display to the left of alert content  */
  icon: PropTypes.element.isRequired,
};
