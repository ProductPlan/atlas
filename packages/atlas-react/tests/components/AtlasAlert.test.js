import React from 'react';
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import AtlasAlert, {alertTypes} from '@components/AtlasAlert'

describe("AtlasAlert", () => {
    it("renders alert content", () => {
    const { container } = render(
        <AtlasAlert
        content="This is a default alert."
        icon={<i className="fas fa-circle-info"></i>}
        />
    )
    const content = screen.getByText("This is a default alert.")
    expect(container).toMatchSnapshot();
    expect(content).toBeTruthy();
    })

    it("renders alert title", () => {
        const { container } = render(
            <AtlasAlert
            title="Finding Nemo"
            content="Just keep swimming."
            icon={<i className="fas fa-circle-info"></i>}
            />
        )
        const title = screen.getByText("Finding Nemo")
        expect(container).toMatchSnapshot();
        expect(title).toBeTruthy();
        })

    it("renders a button", async () => {
        const handleClick = jest.fn();
        const {container} = render(
            <AtlasAlert
            content="This is the content."
            icon={<i className="fas fa-circle-info"></i>}
            label='Click Me!'
            handleClick = {handleClick}
        />
        )
        expect(container).toMatchSnapshot()
        await waitFor(()=> expect(screen.getByText("Click Me!")).toBeTruthy())
        fireEvent.click(screen.getByText("Click Me!"))
        expect(handleClick).toHaveBeenCalled();
    })

    describe("when using different types", () => {
        alertTypes.forEach((type) => {
            it(`renders a ${type} alert`, () => {
                const {container} = render(
                    <AtlasAlert content={`This is a ${type} alert.`}
                    icon={<i className="fas fa-circle-info"></i>} />
                )
                expect(container).toMatchSnapshot();
            })
        })
    })
})