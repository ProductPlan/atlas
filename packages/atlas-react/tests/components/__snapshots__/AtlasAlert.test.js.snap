// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtlasAlert renders a button 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is the content.
      </div>
      <atlas-basic-button
        align="center"
        label="Click Me!"
        size="mini"
        theme="default"
        type="button"
      />
    </div>
  </div>
</div>
`;

exports[`AtlasAlert renders alert content 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is a default alert.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasAlert renders alert title 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__title"
      >
        Finding Nemo
      </div>
      <div
        class="AtlasAlert__content"
      >
        Just keep swimming.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasAlert when using different types renders a danger alert 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is a danger alert.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasAlert when using different types renders a default alert 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is a default alert.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasAlert when using different types renders a information alert 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is a information alert.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasAlert when using different types renders a success alert 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is a success alert.
      </div>
    </div>
  </div>
</div>
`;

exports[`AtlasAlert when using different types renders a warning alert 1`] = `
<div>
  <div
    class="AtlasAlert AtlasAlert--default"
  >
    <i
      class="fas fa-circle-info"
    />
    <div
      class="AtlasAlert__body"
    >
      <div
        class="AtlasAlert__content"
      >
        This is a warning alert.
      </div>
    </div>
  </div>
</div>
`;
